### ISSUE TITLE
`eth_sendTransaction` returns `AlreadyKnown` error, should return transaction hash.

### ISSUE THREAD

**Issue by MicahZoltu** [NethermindEth/nethermind#4148]
**Describe the bug**
If you `eth_sendRawTransaction` for a transaction that is already in the mempool you receive an error of `AlreadyKnown`.  Instead, you should receive the transaction hash.

This is especially impactful when combined with other bugs like when you call `eth_sendRawTransaction` and get back a `FeeTooLow` error (no transaction hash) and then when you send it again you get `AlreadyKnown` error.  This means you never can receive the transaction hash for the transaction in question (unless you calculate it yourself) so you can't actually query it.

If the transaction is already in the mempool, a more appropriate response would be to return the transaction hash which would make `eth_sendRawTransaction` semi-idempotent (you could call it multiple times and get the same answer each time).

**To Reproduce**
1. Via JSON-RPC call `eth_sendRawTransaction` with some transaction.
2. Call `eth_sandRawTransaction` again with the same transaction details.
3. Notice that you receive `AlreadyKnown` error.

**Expected behavior**
You get the transaction hash back.

**Desktop (please complete the following information):**
 - OS: Docker
 - Browser: N/A
 - Version: 1.12.4

**Additional context**
I'm not sure what Geth does here, but even if it returns `AlreadyKnown` I think this is incorrect behavior since you *could* just return the transaction hash which would always be a correct response in this case.

**Comment by LukaszRozmej** [comment]
Geth also sends AlreadyKnown:
https://github.com/ethereum/go-ethereum/issues/22093
https://github.com/ethereum/go-ethereum/issues/22093

**Comment by MicahZoltu** [comment]
I think this is incorrect behavior, given that the send is functionally idempotent.  Is there anything that can be done to get Nethermind to be the first mover on fixing this "bug", or will you only change this if Geth changes it first?

**Comment by LukaszRozmej** [comment]
Yes, I agree it's controversial as the user doesn't care and would prefer not to handle this edge case. We will try changing this.

**Comment by svlachakis** [comment]
Seems outdated, feel free to reopen if it's still an issue
